internal/pkg/util/yamlproto.go

```diff
@@ -0,0 +1,20 @@
+package yamlproto
+
+import (
+	"github.com/ghodss/yaml"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+)
+
+// FromYAMLToProto unmarshals a YAML string into a proto message.
+func FromYAMLToProto(yml string, pb proto.Message) error {
+	js, err := yaml.YAMLToJSON([]byte(yml))
```

Comment:
sorry for bringing this up late, but i think this is a good place to call the `.Validate()` on KeyerConfiguration proto. WDYT?

Reply:
this is a good idea, but I don't think we can do this generally (i.e. there is no `Validate` method in `proto.Message`), which means that we're going to have specialized functions per proto type.

Reply:
Initially I thought that putting the burden on the callers would be a reasonable tradeoff, but on second thought it's better to sanitize protos as early as possible.

Reply:
may be FromYAMLToProto should take a `KeyerConfiguration` proto so that we can call Validate() ? Is there another place you think the validation can be done?

Reply:
I remember @jessicayuen mentioning that we should be able to unmarshal other types, but I might be misremembering. In any case, having a specific function for `KeyerConfiguration` is future-proof enough.

Reply:
Yeah. I think in xds-relay the input file is going to be a `KeyerConfiguration` always. So might be better to solve for that type explicitly.